integration/heapster_api_test.go

```diff
@@ -419,34 +417,195 @@ func runSinksTest(fm kubeFramework, svc *kube_api.Service) error {
+		glog.V(2).Infof("response body: %v", string(body))
+		return nil, err
+	}
+	if err := checkMetricResultListSanity(&data); err != nil {
+		return nil, err
+	}
+	return &data, nil
+}
+
+func getMetricResult(fm kubeFramework, svc *kube_api.Service, url string) (*api_v1.MetricResult, error) {
+	body, err := getDataFromProxy(fm, svc, url)
+	if err != nil {
+		return nil, err
+	}
+	var data api_v1.MetricResult
+	if err := json.Unmarshal(body, &data); err != nil {
+		glog.V(2).Infof("response body: %v", string(body))
+		return nil, err
+	}
+	if err := checkMetricResultSanity(&data); err != nil {
+		return nil, err
+	}
+	return &data, nil
+}
+
+func getStringResult(fm kubeFramework, svc *kube_api.Service, url string) ([]string, error) {
+	body, err := getDataFromProxy(fm, svc, url)
+	if err != nil {
+		return nil, err
+	}
+	var data []string
+	if err := json.Unmarshal(body, &data); err != nil {
+		glog.V(2).Infof("response body: %v", string(body))
+		return nil, err
+	}
+	if len(data) == 0 {
+		return nil, fmt.Errorf("empty string array")
+	}
+	return data, nil
+}
+
+func getStatsResponse(fm kubeFramework, svc *kube_api.Service, url string) (*api_v1.StatsResponse, error) {
+	body, err := getDataFromProxy(fm, svc, url)
+	if err != nil {
+		return nil, err
+	}
+	var data api_v1.StatsResponse
+	if err := json.Unmarshal(body, &data); err != nil {
+		glog.V(2).Infof("response body: %v", string(body))
+		return nil, err
+	}
+	if len(data.Stats) == 0 {
+		return nil, fmt.Errorf("empty stats")
+	}
+	return &data, nil
+}
+
+func getEntityListEntry(fm kubeFramework, svc *kube_api.Service, url string) ([]model.EntityListEntry, error) {
+	body, err := getDataFromProxy(fm, svc, url)
+	if err != nil {
+		return nil, err
+	}
+	var data []model.EntityListEntry
+	if err := json.Unmarshal(body, &data); err != nil {
+		glog.V(2).Infof("response body: %v", string(body))
+		return nil, err
+	}
+	if len(data) == 0 {
+		return nil, fmt.Errorf("empty data")
+	}
+	return data, nil
+}
+
+func checkMetricResultSanity(metrics *api_v1.MetricResult) error {
+	if len(metrics.Metrics) == 0 {
+		return fmt.Errorf("empty metrics")
+	}
+	if time.Now().Sub(metrics.LatestTimestamp).Seconds() > 120 {
+		return fmt.Errorf("corrupted last timestamp")
+	}
+	if time.Now().Sub(metrics.Metrics[0].Timestamp).Seconds() > 120 {
+		return fmt.Errorf("corrupted timestamp")
+	}
+	if metrics.Metrics[0].Value > 10000 {
+		return fmt.Errorf("value too big")
+	}
+	return nil
+}
+
+func checkMetricResultListSanity(metrics *api_v1.MetricResultList) error {
+	if len(metrics.Items) == 0 {
+		return fmt.Errorf("empty metrics")
+	}
+	for _, item := range metrics.Items {
+		err := checkMetricResultSanity(&item)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 func runModelTest(fm kubeFramework, svc *kube_api.Service) error {
-	podList, err := fm.GetPodList()
+	podList, err := fm.GetRunningPods()
 	if err != nil {
 		return err
 	}
-	if len(podList.Items) == 0 {
+	if len(podList) == 0 {
```

Comment:
I'd create a dummy pod here rather than assume that at least one is running in the cluster.